### PR TITLE
Refactor lexer into module & upgrade lexer & parser dependencies

### PR DESCRIPTION
**Pull request by jpschorr**
---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-parser/src/lexer/embedded_ion.rs [hunk 1]
```diff
@@ -0,0 +1,136 @@
+use crate::error::LexError;
+use crate::lexer::{CommentLexer, SpannedResult};
+use logos::{Logos, Span};
+use partiql_common::syntax::line_offset_tracker::LineOffsetTracker;
+use partiql_common::syntax::location::ByteOffset;
+use std::borrow::Cow;
```
Comment (am357) [inline partiql-parser/src/lexer/embedded_ion.rs:6]:
```
 GitHub Actions / clippy

unused import: `std::borrow::Cow`

warning: unused import: `std::borrow::Cow`
 --> partiql-parser/src/lexer/embedded_ion.rs:6:5
  |
6 | use std::borrow::Cow;
  |     ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
```

Comment (jpschorr) [reply partiql-parser/src/lexer/embedded_ion.rs]: removed; not sure why clippy didn't automatically...

**Review by am357 — COMMENTED** [review]
Does this need `CHANGELOG` update?

**Comment by jpschorr** [comment]
> Does this need `CHANGELOG` update?

There are no user-visible API changes.
Parsing hasn't changed.
There are no new features.
There *may* be slight changes to types and bounds of *lexical* errors reported....

**Review by am357 — APPROVED** [review]

**Pull request merged**